mesh/meshdb.go

```diff
@@ -318,6 +332,36 @@ func (m *MeshDB) getLayerMutex(index types.LayerID) *layerMutex {
 	return ll
 }
 
+func getRewardKey(l types.LayerID, account types.Address) []byte {
+	str := "reward_" + account.String() + "_" + strconv.FormatUint(l.Uint64(), 10)
+	return []byte(str)
+}
+
+func getRewardSearchKey(account types.Address) []byte {
+	str := "reward_" + account.String()
+	return []byte(str)
+}
+
+func getTransactionOriginKey(l types.LayerID, t *types.Transaction) []byte {
+	str := "a_orig_" + t.Origin().String() + "_" + strconv.FormatUint(l.Uint64(), 10) + "_" + t.Id().String()
+	return []byte(str)
+}
+
+func getTransactionDestKey(l types.LayerID, t *types.Transaction) []byte {
+	str := "a_dest_" + t.InnerTransaction.Recipient.String() + "_" + strconv.FormatUint(l.Uint64(), 10) + "_" + t.Id().String()
+	return []byte(str)
+}
```

Comment:
I would use the methods generating a prefix and append `t.Id().String()` to it. I think it would be clearer.